services/docker/docker-size.service.js

```diff
@@ -0,0 +1,134 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const prettyBytes = require('pretty-bytes')
+const { nonNegativeInteger } = require('../validators')
+const { latest } = require('../version')
+const {
+  buildDockerUrl,
+  getDockerHubUser,
+  getMultiPageData,
+} = require('./docker-helpers')
+const { NotFound } = require('..')
+const { BaseJsonService } = require('..')
+
+const buildSchema = Joi.object({
+  name: Joi.string().required(),
+  full_size: nonNegativeInteger.required(),
+}).required()
+
+const pagedSchema = Joi.object({
+  count: nonNegativeInteger.required(),
+  results: Joi.array().items(
+    Joi.object({
+      name: Joi.string().required(),
+      full_size: nonNegativeInteger.required(),
+    })
+  ),
+}).required()
+
+const queryParamSchema = Joi.object({
+  sort: Joi.string()
+    .valid('date', 'semver')
+    .default('date'),
+}).required()
+
+module.exports = class DockerSize extends BaseJsonService {
+  static get category() {
+    return 'size'
+  }
+
+  static get route() {
+    return { ...buildDockerUrl('image-size', true), queryParamSchema }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Docker Image Size (latest by date)',
+        pattern: ':user/:repo',
+        namedParams: { user: 'fedora', repo: 'apache' },
```

Comment:
Let's include `queryParams` here explicitly so it will show up in the Shields.io UI for users
```suggestion
       namedParams: { user: 'fedora', repo: 'apache' },
       queryParams: { sort: 'date' },
```

Reply:
I suspect you want the same for the version badge too?

Reply:
Good catch!